lib/widgets/employee_card_list.dart

```diff
@@ -1,68 +1,64 @@
 import 'package:cobe_hive_mobile_app/user_ui_model.dart';
+import 'package:cobe_hive_mobile_app/widgets/chip_list.dart';
 import 'package:cobe_hive_mobile_app/widgets/employee_card.dart';
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:cobe_hive_mobile_app/user_filter_provider.dart';
+
+final statusFilterMap = {
+  UserStatus.online: StatusFilter.online,
+  UserStatus.offline: StatusFilter.offline,
+  UserStatus.sick: StatusFilter.sick,
+  UserStatus.away: StatusFilter.away,
+  UserStatus.vacation: StatusFilter.vacation,
+  UserStatus.parental: StatusFilter.parental,
+  UserStatus.other: StatusFilter.other,
+};
 
 class EmployeeCardList extends StatelessWidget {
   const EmployeeCardList({super.key});
-  //listt of UserUiModel
-  //const EmployeeCardList({required this.employees, super.key});
 
   @override
   Widget build(BuildContext context) {
-    final List<UserUiModel> users = [
-      UserUiModel(
-        name: 'Wade ',
-        surname: 'Warren',
-        status: UserStatus.online,
-        description: 'Text here',
-      ),
-      UserUiModel(
-        name: 'Marry ',
-        surname: 'Roe',
-        status: UserStatus.offline,
-        description: 'Text here',
-      ),
-      UserUiModel(
-        name: 'Tom ',
-        surname: 'Smith',
-        status: UserStatus.parental,
-        description: 'Text here',
-      ),
-      UserUiModel(
-        name: 'Morgan ',
-        surname: 'Reed',
-        status: UserStatus.sick,
-        description: 'Text here',
-      ),
-      UserUiModel(
-        name: 'Jane ',
-        surname: 'Cooper',
-        status: UserStatus.vacation,
-        description: 'Text here',
-      ),
-      UserUiModel(
-        name: 'Arlene ',
-        surname: 'McCoy',
-        status: UserStatus.away,
-        description: 'Text here',
-      ),
-      UserUiModel(
-        name: 'Jonn ',
-        surname: 'Doe',
-        status: UserStatus.other,
-        description: 'Text here',
-      ),
-    ];
+    final userFilter = Provider.of<UserFilterProvider>(context);
+    final searchFilter = Provider.of<UserSearchProvider>(context);
+
+    List<UserUiModel> filteredUsers = UserListTest.users.where(
+      (user) {
+        if (userFilter.isFilterSelected(StatusFilter.all) ||
+            userFilter.isFilterEmpty()) {
+          return true;
+        }
+
+        final userStatusFilter = statusFilterMap[user.status];
+        return userFilter.isFilterSelected(userStatusFilter!);
+      },
+    ).toList();
+
+    List<UserUiModel> searchedUsers = filteredUsers.where(
+      (user) {
+        if (searchFilter.searchQuery.isEmpty) {
+          return true;
+        }
+        if (user.name.toLowerCase().contains(searchFilter.searchQuery) ||
+            user.surname.toLowerCase().contains(searchFilter.searchQuery)) {
+          return true;
+        }
+        return false;
+      },
+    ).toList();
```

Comment:
You can move this logic to another Provider that provides the filtered list of users. You can acomplish this using ProxyProvider [#](https://pub.dev/packages/provider#proxyprovider)